### PR TITLE
Update GP Kron notebook with PyMC version 4

### PR DESCRIPTION
**Pull request by danhphan**
This PR updates the GP Kron notebook from PyMC3 to PyMC version 4. It addresses the issue https://github.com/pymc-devs/pymc-examples/issues/71.

Cc @bwengals and @OriolAbril 

Thank you.

### CODE REVIEW

**Comment by danhphan** [comment]
<p>Hi, yes sure, I will remove it. </p><p> </p><p>I may wait a bit for further comments/feedbacks, and will update these suggestions in one shot. Thanks!</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/455/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by danhphan** [comment]
Hi, I have updated the notebook. Need one more reviewer please. Thank you!

**Comment by danhphan** [comment]
<p>Thanks Bill, I have updated the links of GP classes and watermark :)</p>

File: myst_nbs/gaussian_processes/GP-Kron.myst.md [hunk 1]
```diff
@@ -241,11 +275,33 @@ axs = axs.ravel()
...
+        marker="s",
+        norm=norm,
+        cmap=cmap,
+    )
     ax.set_title(f"Sample {i+1}", fontsize=24)
 ```
 
+## Authors
+* Authored by [Bill Engels](https://github.com/bwengals), 2018
+* Updated by [Chris Fonnesbeck](https://github.com/fonnesbeck), [Raul-ing Average](https://github.com/CloudChaoszero), [Christopher Krapu](https://github.com/ckrapu), [Hector Munoz](https://github.com/hectormz), and [Danh Phan](https://github.com/danhphan), 2020-2022
```
Comment (OriolAbril) [inline myst_nbs/gaussian_processes/GP-Kron.myst.md:292]: From a quick look at the [history](https://github.com/pymc-devs/pymc-examples/commits/main/examples/gaussian_processes/GP-Kron.ipynb) of the notebook after it was moved to pymc-examples, according to what we agreed on with https://docs.pymc.io/en/latest/contributing/jupyter_style.html#authorship-and-attribution and https://docs.pymc.io/en/latest/contributing/jupyter_style.html#first-cell (last sentence in the section, relevant for adding people to author section in post directive). This should be something like

```
## Authors
* Authored by [Bill Engels](https://github.com/bwengals), 2018
* Updated by [Raul-ing Average](https://github.com/CloudChaoszero), March 2021
* Updated by [Christopher Krapu](https://github.com/ckrapu), July 2021
* Updated to PyMC 4.x by [Danh Phan](https://github.com/danhphan), November 2022
```

Chris moved the notebook, which we don't list here, I added the myst counterparts and in the sync process some trailing whitespaces and things like that were modified, and Hector ran codespell which fixed a couple typos.

File: myst_nbs/gaussian_processes/GP-Kron.myst.md [hunk 2]
```diff
@@ -241,11 +275,33 @@ axs = axs.ravel()
...
+
++++
+
+## References
+:::{bibliography}
+:filter: docname in docnames
+:::
```
Comment (OriolAbril) [inline myst_nbs/gaussian_processes/GP-Kron.myst.md:299]: Same comment as in the other notebooks, there are no references using the cite role (as shown in https://docs.pymc.io/en/latest/contributing/jupyter_style.html#references) so this shoud be omited. The watermark section title is missing below though

**Comment by danhphan** [comment]
Hi, thanks @OriolAbril. The authors and watermark of this notebook have been updated.

**Review by OriolAbril — APPROVED** [review]

**Pull request merged**